netboot/netconf.go

```diff
@@ -127,5 +217,36 @@ func ConfigureInterface(ifname string, netconf *NetConf) error {
 		resolvconf += fmt.Sprintf("nameserver %s\n", ns)
 	}
 	resolvconf += fmt.Sprintf("search %s\n", strings.Join(netconf.DNSSearchList, " "))
-	return ioutil.WriteFile("/etc/resolv.conf", []byte(resolvconf), 0644)
+	if err = ioutil.WriteFile("/etc/resolv.conf", []byte(resolvconf), 0644); err != nil {
+		return fmt.Errorf("could not write resolv.conf file %v", err)
+	}
+
+	iface, err = netlink.LinkByName(ifname)
+	if err != nil {
+		return fmt.Errorf("could not obtain interface when adding default route: %v", err)
+	}
+
+	dst := &net.IPNet{
+		IP:   net.IPv4(0, 0, 0, 0),
+		Mask: net.CIDRMask(0, 32),
+	}
+
```

Comment:
I wonder about the implications of doing this for a v6 client.

Reply:
As discussed, this can be indeed moved a bit further down where I check if I got a default route from the packet. If that's the case, then I don't have any other option but removing the current default route. If we want to support IPv6 routes via DHCPv6, then the `Netconf ` struct will have to be extended to hold both v4 and v6 routes. I can straight away rename `Routers` to `Routersv4`, if you think there is good chance that the library will get IPv6 routes support.

Reply:
Moving a bit further look good to me.